src/OidcClientSettings.ts

```diff
@@ -90,8 +90,14 @@ export interface OidcClientSettings {
     /**
      * Indicates if objects returned from the user info endpoint as claims (e.g. `address`) are merged into the claims from the id token as a single object.
      * Otherwise, they are added to an array as distinct objects for the claim type. (default: false)
+     * @deprecated since version 2.3.0. It's not a deterministic way of handling claims.
      */
     mergeClaims?: boolean;
```

Comment:
Maybe it makes sense to keep this settings in the long turn in the way it was supposed:

From the description:
```
Indicates if objects returned from the user info endpoint as claims (e.g. `address`) are merged into the claims from the id token as a single object.
Otherwise, they are added to an array as distinct objects for the claim type. (default: false)
```
-> true = merged, false = added/appended as an array.

-> mergeClaims = true -> the new behavior of array handling
-> mergeClaims = false -> the legacy behavior

We keep the default as is and in a major release we change the default to `true`.
The `mergeClaimsLegacyBehavior` is then not needed

What do you think? Does that make sense?

Reply:
If we agree on the above it makes sense to rename `legacyMergeClaims` to `appendClaims` and **not** call it from `mergeClaims`. Instead we make and if/else on the caller side depending on `mergeClaims`.
-> easier to understand and easier to follow the unit-tests.

Reply:
@pamapa frankly I just want this implementation to go ahead.

The current behavior is wrong, wherever we spin it we will have some breaking changes unless we create new attributes to keep this wrong behavior.

Right now `mergeClaims`, when enabled, works as a `deep merging` mechanism, which we will have by default in the new implementation, and changing its meaning will break some usage somewhere.

Whichever solution makes it clearer for you makes sense to me, I'll stick to the new implementation and try to forget about all of this ASAP, but we have 40 comments on this PR and I think we need to land it somehow.

Reply:
@Badisi Should we keep the setting `mergeClaims` in the long run?

Reply:
> I'll stick to the new implementation and try to forget about all of this ASAP

I will help you here. First step see #904

Reply:
I agree with @marcoreni that this whole `mergeClaims` thing is wrong from the start and that it's taking us too much discussion. TBH I don't even recall why we were talking about rewriting it at the first place. I thought that @marcoreni requirement was only to be able to reload UserInfo ?

So do we have any issue today with this merge mechanism ? If not then I would suggest to leave it as is, as it seems to work well for everybody (else we would have had other complains). And if we really need to refactor it I would go with a `mergeClaimsStrategy=['none','merge','replace']` (or something alike), not so disruptive and with all use cases covered.

Reply:
> requirement was only to be able to reload UserInfo

will not work with the current wrong merge behavior...

Reply:
I'm confused..
- The first time, `IdToken` will be merged with `UserInfo` data.
- The second time (when asking to refresh user info), we will use the current `IdToken` and merge it with the newly received `UserInfo` data.

This should be exactly the same.. why it wouldn't work ?

Reply:
@Badisi You are right lets stop this merge request for now. And implement the feature like you said.

Merging claims is not perfect, we can change that later...